master/internal/plugin/oidc/service.go

```diff
@@ -272,19 +309,52 @@ func (s *Service) lookupUser(ctx context.Context, claimValue string) (*model.Use
 	return u, err
 }
 
+func mergeUserGroups(sessionData *IDTokenClaims, dbData *model.AgentUserGroup) *model.AgentUserGroup {
+	result := model.AgentUserGroup{
+		UID:   dbData.UID,
+		GID:   dbData.GID,
+		User:  dbData.User,
+		Group: dbData.Group,
+	}
+
+	if sessionData.AgentUID != 0 {
```

Comment:
If I am reading this correctly, it looks like we initialize all the `result` with the UID and GID that already exist for the user in the DB.

I am guessing that we are checking the AgentUID and AgentGID are not equal to 0 because golang auto initializes ints that aren't assigned values to 0 when in reality we want to check to see if any value, including 0, has been set in the IDTokenClaims.AgentUID and IDTokenClaims.AgentGID.

While this likely won't happen, I imagine a situation in which a user had a specific non-zero UID and GID set for their user. The user is then promoted to a "sys admin" or something and they now need to run tasks as root. If I am reading this code correctly, in this situation we wouldn't update the UID and GID of the user when the auth claim returned contained a AgentUID and AgentGID of 0.

If I am correct, I think we might need to do something like use a int pointer and check if nil vs. checking if 0 to handle situations in which the admin wants to intentionally give someone permission to run as root.

Reply:
Yes, the default value for an int in Golang is 0. You're absolutely right that, as is, it is not possible to reset the agent uid/gid to 0 - great catch! 

I don't know about the plausibility of the specific use case mentioned. I'm going to start a conversation in a team channel so people with more experience with this part of the code can weigh in.

Reply:
Cool, I see a couple of bool vars were added to handle this. Thanks!